From 3fffdbe01cccc8f83482651e7de11f296893108b Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Thu, 11 Apr 2024 14:40:21 +0100
Subject: [PATCH] Do not use bool as a field name

C99 aliases `bool` to `_Bool`, and C23 introduces `bool` as a reserved
keyword. Let's avoid using `bool` as a field name.

Upstream-Status: Backport [Backport from glib to bundled version in pkg-config https://github.com/GNOME/glib/commit/9e320e1c43a4770ed1532248fe5416eb0c618120]
---
 glib/glib/goption.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/glib/goption.c b/glib/glib/goption.c
index 0a22f6f..f439fd4 100644
--- a/glib/glib/goption.c
+++ b/glib/glib/goption.c
@@ -166,7 +166,7 @@ typedef struct
   gpointer arg_data;
   union
   {
-    gboolean bool;
+    gboolean boolean;
     gint integer;
     gchar *str;
     gchar **array;
@@ -1600,7 +1600,7 @@ free_changes_list (GOptionContext *context,
           switch (change->arg_type)
             {
             case G_OPTION_ARG_NONE:
-              *(gboolean *)change->arg_data = change->prev.bool;
+              *(gboolean *)change->arg_data = change->prev.boolean;
               break;
             case G_OPTION_ARG_INT:
               *(gint *)change->arg_data = change->prev.integer;
-- 
2.50.1

